tests/testsuite/git.rs

```diff
@@ -3760,3 +3762,89 @@ fn different_user_relative_submodules() {
 
     assert!(project.bin("foo").is_file());
 }
+
+#[cargo_test(requires_git)]
+fn cargo_package_with_new_worktree() {
+    let project = project().build();
+    let git_project = git::new("foo", |project| {
+        project.file("src/lib.rs", "").file("README.md", "").file(
+            "Cargo.toml",
+            &format!(
+                r#"[package]
+name = "foo"
+version = "0.1.0"
+edition = "2015"
+authors = []
+license = "MIR OR Apache-2.0"
+description = "A test!"
+homepage = "https://example.org"
+documentation = ""
+repository = "https://example.org"
+readme = "./README.md"
+"#,
+            ),
+        )
+    });
+
+    let repo = git2::Repository::open(&git_project.root()).unwrap();
+    let repo_root = repo.workdir().unwrap().parent().unwrap();
+    let opts = git2::WorktreeAddOptions::new();
+    let wt = repo
+        .worktree("bar", &repo_root.join("bar"), Some(&opts))
+        .unwrap();
+
+    project
+        .cargo("package --list")
+        .cwd(wt.path())
+        .with_stdout(
+            "\
+.cargo_vcs_info.json
+Cargo.toml
+Cargo.toml.orig
+README.md
+src/lib.rs
+",
+        )
+        .run();
+}
+
+#[cargo_test(requires_git)]
+fn cargo_check_with_sparse_checkouts() {
```

Comment:
Could you remove this test?

libgit2 hasn't supported sparse checkout (see <https://github.com/rust-lang/cargo/issues/11165>), hence cargo doesn't. I believe this test work just because the repo already has everything checked out.

Reply:
deleted。 🆗